core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -483,7 +483,24 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
               provide(fullyQualifiedActionName(actionName)) { fullActionName =>
                 onComplete(verifyWebAction(fullActionName, onBehalfOf.isDefined)) {
                   case Success((actionOwnerIdentity, action)) =>
-                    if (!action.annotations.getAs[Boolean]("web-custom-options").exists(identity)) {
+                    val requireWebAuthIsBool = (action.annotations.getAs[Boolean]("require-whisk-auth") != None)
+                    val annotationRequireWebAuthIsIntOrString = ((action.annotations
+                      .getAs[String]("require-whisk-auth") != None) || (action.annotations
+                      .getAs[Int]("require-whisk-auth") != None))
+                    val annotationRequireWebAuth = (action.annotations
+                      .getAs[Int]("require-whisk-auth")
+                      .getOrElse(action.annotations.getAs[String]("require-whisk-auth").getOrElse("")))
+                      .toString
+                    val enforceWhiskAuthHdr = (!requireWebAuthIsBool && annotationRequireWebAuthIsIntOrString)
+                    val headerWhiskAuthSeq = context.headers.filter(_.lowercaseName == "x-require-whisk-auth")
```

Comment:
If you want to get only one element which matches the predicate, use `find` over `filter`. It will return an `Option[Header]` which more closely reflects what you're after.

Reply:
yes, that works much better.  thanks!